### PR TITLE
fix combined cards crash when some cards do not have metrics and dimensions settings

### PR DESCRIPTION
**Pull request by alxnddr**
Closes https://github.com/metabase/metabase/issues/41290

### Description

When multiple cards (datasets) are combined on a single dashboard card the settings prop of the visualization contains only one set of metrics and dimensions which by design is not sufficient for multiple cards. At the same time, not all cards settings contain saved "graph.dimensions" and "graph.metrics" so we have to get defaults if they are not present.

### How to verify

This PR updates `e2e/test/scenarios/visualizations-charts/line-bar-tooltips.cy.spec.js` and `e2e/test/scenarios/visualizations-charts/line_chart.cy.spec.js` specs because they contained combined cards which failed before but work now. However, the specs are still failing due to incorrect values in tooltips which I'm going to tackle next.

### Demo

Combined cards chart from `should show updated column titles in tooltips after editing them via Visualization Options` spec:

Screenshot 2024-04-22 at 11 09 33 PM

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/test/scenarios/visualizations-charts/line-bar-tooltips.cy.spec.js
Comment (JesseSDevaney) [inline e2e/test/scenarios/visualizations-charts/line-bar-tooltips.cy.spec.js:1]: - https://github.com/metabase/metabase/blob/fix-combined-cards-crash/e2e/test/scenarios/visualizations-charts/line-bar-tooltips.cy.spec.js#L363

This test was also one the ones failing, but I noticed you did not update that one as well. Would you update that one as well to account for new selectors?

**Review by JesseSDevaney — CHANGES_REQUESTED** [review]
Thank you for fixing dash-cards with added series, Sasha! 

I checked the tests that were blocked by the corresponding issue and all of them are no longer failing because of added series errors - they are still failing but for other reasons. 

But, there is still one chart listed in that issue that is still crashing, see below.

---

The chart from https://github.com/metabase/metabase/issues/41290, shown below, is still erroring out.

Screenshot from 2024-04-22 15-20-19
image

- Relevant test: https://github.com/metabase/metabase/blob/echarts/e2e/test/scenarios/visualizations-charts/bar_chart.cy.spec.js#L406

**Comment by alxnddr** [comment]
@JesseSDevaney nice catch, thanks! I fixed the problem

File: e2e/test/scenarios/visualizations-charts/line-bar-tooltips.cy.spec.js
Comment (alxnddr) [reply e2e/test/scenarios/visualizations-charts/line-bar-tooltips.cy.spec.js:1]: I fixed them here https://github.com/metabase/metabase/pull/41711 and now I'll address your feedback there

**Review by JesseSDevaney — APPROVED** [review]
Thank you for fixing the broken charts! :heart:

**Pull request merged**